### PR TITLE
JSON API: New Options Endpoint

### LINKED ISSUE #2638: JSON API: endpoint for managing arbitrary options
Had a conversation with @beaulebens where we decided that having an endpoint
to fetch an arbitrary option, or a group of arbitrary options would be nice.
Also, there should be a way to modify an arbitrary option.

This will help in building new features on remote applications.

### PR DESCRIPTION
**Pull request by roccotripaldi**
Creates a new endpoint that will fetch an arbitrary site option.

We are still working on this, we also need to create an endpoint to update an option.

Closes #2638 

cc: @enejb @lezama

### CODE REVIEW

**Comment by lezama** [comment]
This is looking good @roccotripaldi!

File: json-endpoints/jetpack/class.jetpack-json-api-options-update-endpoint.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class Jetpack_JSON_API_Options_Update_Endpoint extends Jetpack_JSON_API_Options_Endpoint {
+    // GET /sites/%s/options/$option
+    protected $needed_capabilities = 'manage_options';
+    protected $action = 'update';
+
+    public function update() {
+        $query_args = $this->query_args();
+        $input = $this->input();
+        if( $query_args['site_option'] ) {
```
Comment (beaulebens) [inline json-endpoints/jetpack/class.jetpack-json-api-options-update-endpoint.php]: Space after `if`.

File: json-endpoints/jetpack/class.jetpack-json-api-options-endpoint.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    protected $option_value;
+
+    function result() {
+        return array( 'value' => $this->option_value );
+    }
+
+    protected function validate_input( $option ) {
+        $args = $this->query_args();
+        $this->option_name = $option;
+        if( $args['site_option'] ) {
```
Comment (beaulebens) [inline json-endpoints/jetpack/class.jetpack-json-api-options-endpoint.php]: Space after `if`.

**Comment by roccotripaldi** [comment]
Closing this. We'll be taking a different approach that i'll address in an other PR

**Pull request closed without merging**